tests/serialize/runstate/dynamodb_state_store_test.py

```diff
@@ -125,7 +125,7 @@ def large_object():
         "runs": [],
         "cleanup_run": None,
         "manual": False,
-        "large_data": [i for i in range(1_000_000)],
+        "large_data": [i for i in range(10000)],
```

Comment:
was this meant to be changed?

Reply:
yea, I was getting this error
``
ClientError('An error occurred (ValidationException) when calling the TransactWriteItems operation: Item size has exceeded the maximum allowed size')
``
moto's mock enforces a 400kb item size limit and the 1000000 serializes to a json object greater than 400kb. I think older versions of moto didn't enforce this size limit and now that we upgraded moto it enforces it. So I reduced the number of integers we create here to be below that limit

Reply:
gotcha! i assume that since the tests are still passing, then we're still ending up
splitting this object

Reply:
I suspect that by reducing the data to 10,000 integers, we're now creating an object that is smaller than our OBJECT_SIZE of 200KB. This would mean it no longer gets partitioned, and these tests are no longer verifying the multi-partition logic they were designed for.

Maybe we should also add an assert num_partitions(key) > 1 in these tests to guarantee we're actually testing the partitioning logic?

Also, outside the bounds of this PR, it might be a good time to rename these tests to something like test_save_object_requiring_partitioning. `more_than_4KB` is _technically_ correct but confusing 😄